src/components/editor/EditorArticleController.jsx

```diff
@@ -13,109 +14,143 @@ export default class EditorArticleController extends FalcorController {
   constructor(props) {
     super(props);
     this.handleSaveChanges = this.handleSaveChanges.bind(this);
-    this.mainFormChanged = this.mainFormChanged.bind(this);
-    this.handleMainFormChange = this.handleMainFormChange.bind(this);
-    this.authorListChanged = this.authorListChanged.bind(this);
+    this.handleFormChanges = this.handleFormChanges.bind(this);
     this.safeSetState({
       changed: "none",
       saving: false
     });
+    this.debouncedHandleFormChanges = debounce(this.handleFormChanges, 500);
   }
 
-  static getFalcorPath(params) {
+  static getFalcorPathSets(params) {
     return [['articlesBySlug', params.slug, ['title', 'issue', 'category', 'description']], ['articlesBySlug', params.slug, 'authors', {to: 2}, ['slug', 'name']]];
   }
 
   handleSaveChanges(event) {
     event.preventDefault();
-  }
+    const formNode = event.target;
+    // Here is how data can be retrieved when needed for post requests
+    // Remember to check if it changed
 
-  mainFormChanged(event) {
-    if (event.target.name === "addAuthor") {
-      return;
-    }
+    // This can also be done with an iterator like in handleFormChanges
+    const category = formNode.category.value;
+    const issue = formNode.issue.value;
+    const description = formNode.description.value;
 
-    let changedFlag = this.handleMainFormChange();
+    const authorsAdded = this.editAuthorListNode.state.authorsAdded;
+    const authorsDeleted = this.editAuthorListNode.state.authorsDeleted;
+    // How to retrieve data done
 
-    let newChangedState = changedFlag ? "changed" : "none";
-    if (newChangedState !== this.state.changed) {
-      this.safeSetState({
-        changed: newChangedState
+    // Change data asynchronously
+    this.safeSetState({
+      changed: "saving"
+    });
+    setTimeout(
+      () => {
+        Promise.resolve("success").then(() => {
+          // Just use falcor to change data here instead of window reload
+          this.safeSetState({
+            changed: "saved"
+          });
+          setTimeout(() => {
+            window.location.reload();
+          }, 1000);
       });
-    }
+    }, 2000);
   }
 
-  authorListChanged(authorsAdded, authorsDeleted) {
+  handleFormChanges() {
+    const authorsAdded = this.editAuthorListNode.state.authorsAdded;
+    const authorsDeleted = this.editAuthorListNode.state.authorsDeleted;
+
     let changedFlag = Object.keys(authorsAdded).length > 0 || 
       _.some(authorsDeleted, (val) => {
```

Comment:
Well as of now neither authorsDeleted nor authorsAdded are arrays. They're both objects. authorsDeleted has a true/false value to whether the original authors were already deleted and authorsAdded is going to be a connection between slug and name as we'll call for the authors by slug but display their name. And also what the editors will type in to search for them will be their name, but we need their slug to be able to access their data through falcor. 

I think this works okay right now according to me. I will definitely look over it again when I actually have the database structured and I'm actually doing post requests though, and then I may or may not restructure things quite a bit.